gspread/worksheet.py

```diff
@@ -2607,6 +2607,28 @@ def unmerge_cells(self, name):
 
         return self.spreadsheet.batch_update(body)
 
+    def get_notes(self):
+        """Returns a list of lists containing all notes in the sheet, or the empty list if the
+        sheet does not have a note.
+        """
+        url = SPREADSHEET_URL % (self.spreadsheet.id)
+        params = {"fields": "sheets/data/rowData/values/note"}
+        response = self.client.request("get", url, params=params)
+        response.raise_for_status()
+        response_json = response.json()
+
+        try:
+            data = response_json["sheets"][self.index]["data"][0]["rowData"]
```

Comment:
Up to the field `["sheets"][self.index]["data"]` you don't need to check the API will return it for sure. even if you don't have any notes.

Reply:
in a way yes, what I mean is: you can reach for sure the data up to `["sheets"][self.index]["data"][0]`, the API will always give you that JSON tree at least. Then from there you can use the python notation to get the entry of a map or a default values like:

```python
data = response_json["sheets"][self.index]["data"][0].get("rowData", [])
```
from here you always have a list to iterate over.